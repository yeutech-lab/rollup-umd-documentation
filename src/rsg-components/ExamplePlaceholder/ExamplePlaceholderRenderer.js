import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Styled from 'rsg-components/Styled';
import Markdown from 'rsg-components/Markdown';
import { DOCS_DOCUMENTING } from '../../../scripts/consts';

const styles = ({ fontFamily, fontSize, color }) => ({
	button: {
		padding: 0,
		fontSize: fontSize.base,
		fontFamily: fontFamily.base,
		textDecoration: 'underline',
		color: color.light,
		border: 0,
		cursor: 'pointer',
		background: 'transparent',
		'&:hover, &:active': {
			isolate: false,
			color: color.lightest,
		},
	},
});

export class ExamplePlaceholderRenderer extends Component {
	static propTypes = {
		classes: PropTypes.object.isRequired,
		name: PropTypes.string,
	};

	constructor() {
		super();
		this.handleOpen = this.handleOpen.bind(this);
		this.state = {
			isVisible: false,
		};
	}

	handleOpen() {
		this.setState({ isVisible: true });
	}

	render() {
		const { classes, name } = this.props;
		const { isVisible } = this.state;
		if (isVisible) {
			return (
				<Markdown
					text={`
Create **Readme.md** or **${name}.md** file in the componentâ€™s folder like this:

    ${name} example:

    \`\`\`js
    <${name} pizza="\uD83C\uDF55" />
	\`\`\`

You may need to **restart** the style guide server after adding an example file.

Read more in the [documenting components guide](${DOCS_DOCUMENTING}).
					`}
				/>
			);
		}

		return (
			<button className={classes.button} onClick={this.handleOpen}>
				Add examples to this component
			</button>
		);
	}
}

export default Styled(styles)(ExamplePlaceholderRenderer);
