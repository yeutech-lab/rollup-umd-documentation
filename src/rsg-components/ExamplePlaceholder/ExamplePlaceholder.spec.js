import React from 'react';
import { ExamplePlaceholderRenderer } from './ExamplePlaceholderRenderer';

it('should render a link', () => {
	const actual = shallow(<ExamplePlaceholderRenderer classes={{}} name="Pizza" />);

	expect(actual).toMatchSnapshot();
});

it('should render an example placeholder after click', () => {
	const actual = shallow(<ExamplePlaceholderRenderer classes={{}} name="Pizza" />);

	actual.find('button').simulate('click');

	expect(actual).toMatchSnapshot();
});
