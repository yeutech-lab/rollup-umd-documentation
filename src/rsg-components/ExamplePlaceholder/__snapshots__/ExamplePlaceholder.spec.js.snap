// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a link 1`] = `
<button
  onClick={[Function]}
>
  Add examples to this component
</button>
`;

exports[`should render an example placeholder after click 1`] = `
<Markdown
  text="
Create **Readme.md** or **Pizza.md** file in the component’s folder like this:

    Pizza example:

    \`\`\`js
    <Pizza pizza=\\"🍕\\" />
	\`\`\`

You may need to **restart** the style guide server after adding an example file.

Read more in the [documenting components guide](https://react-styleguidist.js.org/docs/documenting.html).
					"
/>
`;
